### PR TITLE
HerbGrammar `v0.4.0`

### PR DESCRIPTION
**Pull request by ReubenJ**
TODO

- [x] Update documentation for `RuleNode._val` (https://github.com/Herb-AI/HerbCore.jl/pull/25)

### Breaking Changes

- #79 
- #75 

### Bug Fixes

- #84 

### Performance

- #82

### CODE REVIEW

File: src/csg/probabilistic_csg.jl [hunk 1]
```diff
@@ -36,17 +36,6 @@ function expr2pcsgrammar(ex::Expr)::ContextSensitiveGrammar
 		end
 	end
 	alltypes = collect(keys(bytype))
-	# Normalize probabilities for each type
```
Comment (THinnerichs) [inline src/csg/probabilistic_csg.jl:39]: Why are we removing normalization of probabilities again?

File: src/csg/csg.jl [hunk 2]
```diff
@@ -116,7 +116,7 @@ end
 - [`@pcsgrammar`](@ref) uses a similar syntax to create probabilistic [`ContextSensitiveGrammar`](@ref)s.
 """
 macro csgrammar(ex)
-	return expr2csgrammar(ex)
+	return :(expr2csgrammar($(QuoteNode(ex))))
```
Comment (THinnerichs) [inline src/csg/csg.jl:119]: Could you describe/link the issue which this is solving? In what cases do we not get an expression here?

File: src/csg/probabilistic_csg.jl (hunk 1 above)
Comment (ReubenJ) [reply src/csg/probabilistic_csg.jl:39]: We already do it at the end of this method:

https://github.com/Herb-AI/HerbGrammar.jl/blob/671a3024cf6edfdb78045542b12fa47865f4a22e/src/csg/probabilistic_csg.jl#L46

File: src/csg/csg.jl (hunk 2 above)
Comment (ReubenJ) [reply src/csg/csg.jl:119]: In all cases. `expr2csgrammar(ex)` returns a `ContextSensitiveGrammar` object. Example (HerbGrammar v0.3.0):

```julia
julia> res = @macroexpand @csgrammar begin
               A = 1
       end
1: A = 1


julia> typeof(res)
ContextSensitiveGrammar
```

With this change:

```julia
julia> res = @macroexpand @csgrammar begin
           A = 1
       end
:(HerbGrammar.expr2csgrammar($(QuoteNode(quote
    #= REPL[4]:2 =#
    A = 1
end))))

julia> typeof(res)
Expr

julia>
```

This makes it possible to set a breakpoint in the `expr2csgrammar` method and use a debugger. The debugger (at least in VSCode) doesn't break on any code that runs within the macro expansion step.

Generally, the pattern for macros is to modify the code and return the new `Expr`. See `@show`:

```julia
julia> x = "Hello, world!"
"Hello, world!"

julia> res = @macroexpand @show x
quote
    Base.println("x = ", Base.repr(begin
                #= show.jl:1181 =#
                local var"#3#value" = x
            end))
    var"#3#value"
end

julia> typeof(res)
Expr
```

**Review by THinnerichs — APPROVED** [review]

**Pull request merged**